File: day6_2/src/direction.rs
#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash)]
pub enum Direction {
    Up,
    Right,
    Down,
    Left,
}

impl Direction {
    pub fn rotate_right(&mut self) {
        *self = match *self {
            Direction::Up => Direction::Right,
            Direction::Right => Direction::Down,
            Direction::Down => Direction::Left,
            Direction::Left => Direction::Up,
        }
    }
}

File: day06_2/py/direction.py
from enum import Enum

class Direction(Enum):
    Up = 1
    Right = 2
    Down = 3
    Left = 4

    def rotated_right(self) -> "Direction":
        match self:
            case Direction.Up: return Direction.Right
            case Direction.Right: return Direction.Down
            case Direction.Down: return Direction.Left
            case Direction.Left: return Direction.Up
